packages/core-components/src/layout/SignInPage/SignInPage.tsx

```diff
@@ -105,47 +106,42 @@ export const SingleSignInPage = ({
   type LoginOpts = { checkExisting?: boolean; showPopup?: boolean };
   const login = async ({ checkExisting, showPopup }: LoginOpts) => {
     try {
-      let identity: BackstageIdentity | undefined;
+      let identityResponse: BackstageIdentityResponse | undefined;
       if (checkExisting) {
         // Do an initial check if any logged-in session exists
-        identity = await authApi.getBackstageIdentity({
+        identityResponse = await authApi.getBackstageIdentity({
           optional: true,
         });
       }
 
       // If no session exists, show the sign-in page
-      if (!identity && (showPopup || auto)) {
+      if (!identityResponse && (showPopup || auto)) {
         // Unless auto is set to true, this step should not happen.
         // When user intentionally clicks the Sign In button, autoShowPopup is set to true
         setShowLoginPage(true);
-        identity = await authApi.getBackstageIdentity({
+        identityResponse = await authApi.getBackstageIdentity({
           instantPopup: true,
         });
-        if (!identity) {
+        if (!identityResponse) {
           throw new Error(
             `The ${provider.title} provider is not configured to support sign-in`,
           );
         }
       }
 
-      if (!identity) {
+      if (!identityResponse) {
         setShowLoginPage(true);
         return;
       }
 
       const profile = await authApi.getProfile();
-      onResult({
-        userId: identity!.id,
-        profile: profile!,
-        getIdToken: () => {
-          return authApi
-            .getBackstageIdentity()
-            .then(i => i!.token ?? i!.idToken);
-        },
-        signOut: async () => {
-          await authApi.signOut();
-        },
-      });
+      onSignInSuccess(
+        UserIdentity.from({
```

Comment:
Reading this it does look a bit weird tbh, thinking maybe one of these?

```ts
UserIdentity.create({
  identity: identityResponse.indentity,
  authApi,
  profile
})
```

or

```ts
UserIdentity.fromIdentity(identityResponse.indentity, {
  authApi,
  profile
})
```

I think I like `.create` in this case more though